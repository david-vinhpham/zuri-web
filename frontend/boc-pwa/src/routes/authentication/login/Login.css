@import '../../../components/variables.css';

.container {
  width: 100%;
  height: 100%;
  position: absolute;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: flex-start;
  background: radial-gradient(circle, rgba(255, 255, 255, 1) 0%, rgba(147, 226, 255, 1) 100%);
}

.welcome {
  color: var(--c-brand-3);
  text-align: center;
}

.logo {
  margin: var(--size-64) auto;
  width: var(--home-logo-width);
}

.title {
  margin: var(--size-12) 0 var(--size-36);
}

.brand {
  width: var(--home-brand-width);
  margin: var(--size-14);
}

.greeting {
  color: var(--c-brand-1);
  font-size: var(--font-size-16);
}

.login {
  display: flex;
  position: absolute;
  bottom: calc(var(--size-78) * 1.3);
}

.button {
  color: var(--c-text-bright);
  background: var(--c-brand-1);
}

.register {
  position: absolute;
  color: var(--c-brand-2);
  font-size: var(--font-size-14);
  bottom: var(--size-64);
}

.register span {
  text-transform: uppercase;
  font-weight: var(--font-weight-500);
  cursor: pointer;
  color: var(--c-error);
}

.register span:hover {
  font-weight: var(--font-weight-500);
}

.copyRight {
  position: absolute;
  bottom: var(--size-12);
  color: var(--c-brand-1);
  font-size: var(--font-size-12);
  text-align: center;
}
