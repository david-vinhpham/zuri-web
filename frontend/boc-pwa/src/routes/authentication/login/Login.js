import React from 'react';
import withStyles from 'isomorphic-style-loader/lib/withStyles';
import BocLogo from 'assets/images/welcome_boc.png';
import BocGreet from 'assets/images/boc_greeting.png';
import { HowToReg } from 'constants/svg';
import Button from 'components/Button';
import history from '../../../history';
import s from './Login.css';

class Login extends React.Component {
  handleActivation = () => {
    history.push('/activation');
  };
  redirectRegister = () => {
    history.push('/activation', {
      register: true,
    });
  };

  render() {
    return (
      <div className={s.container}>
        <div className={s.welcome}>
          <img src={BocLogo} alt="boc-logo" className={s.logo} />
          <div className={s.title}>
            <img src={BocGreet} alt="Boc Brand" className={s.brand} />
            <div className={s.greeting}>Quản lý cửa hàng hiệu quả hơn</div>
          </div>
        </div>
        <div className={s.login}>
          <Button
            label="Đăng Nhập"
            className={s.button}
            onClick={this.handleActivation}
          >
            <HowToReg />
          </Button>
        </div>
        <div className={s.register}>
          Hoặc chưa có tài khoản?{' '}
          <span onClick={this.redirectRegister}>Đăng ký</span>
        </div>
        <div className={s.copyRight}>Bản quyền thuộc về BOCVN@2019</div>
      </div>
    );
  }
}

export default withStyles(s)(Login);
