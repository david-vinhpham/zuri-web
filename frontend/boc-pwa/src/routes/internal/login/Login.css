@import '../../../components/variables.css';

:root {
  --title-size: 14px;
  --title-weight: 700;
  --container-p: 16px;
  --btn-transition-duration: 200ms, 200ms, 200ms, 0.3s;
  --login-width: 35rem;
  --login-border: 1px solid var(--c-ui-light);
}

.container {
  padding: var(--container-p);
  text-align: center;
}

.submitBtn {
  display: none;
}

.title {
  font-weight: var(--title-weight);
  font-size: var(--title-size);
  color: var(--c-text);
  font-style: italic;
  text-align: center;
}

.groupBtn {
  text-align: center;
  padding-top: var(--b-space-md);
}

@media (--sm-min) {
  .login {
    display: inline-block;
    width: var(--login-width);
    padding: var(--b-space-xl) var(--b-space-3xl);
    border: var(--login-border);
  }
}
