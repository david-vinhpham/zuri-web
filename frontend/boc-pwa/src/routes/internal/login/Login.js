import React, { PureComponent } from 'react';
import { Input, Button } from 'homecredit-ui';
import { withFormik } from 'formik';
import { compose } from 'redux';
import { connect } from 'react-redux';
import { func, shape, bool, string } from 'prop-types';
import withStyles from 'isomorphic-style-loader/lib/withStyles';
import noop from 'lodash/noop';
import isEmpty from 'lodash/isEmpty';

import { getValidationSchema } from 'utils/validations';
import { login, logout } from 'actions/internal';
import history from '../../../history';
import s from './Login.css';

import {
  LOGIN_TITLE,
  USERNAME_PLACEHOLDER,
  PASSWORD_PLACEHOLDER,
  LOGIN_BUTTON,
} from './constant';

class Login extends PureComponent {
  static propTypes = {
    values: shape({
      username: string,
      password: string,
    }),
    loginAction: func,
    logout: func,
    touched: shape({
      username: bool,
      password: bool,
    }),
    errors: shape({
      username: string,
      password: string,
    }),
    handleChange: func,
    handleBlur: func,
    validateForm: func.isRequired,
    setFieldTouched: func.isRequired,
  };

  static defaultProps = {
    values: {},
    loginAction: noop,
    logout: noop,
    touched: {},
    errors: {},
    handleChange: noop,
    handleBlur: noop,
  };

  componentWillMount() {
    this.props.logout(null);
  }

  handleChange = e => {
    const { handleChange } = this.props;
    handleChange(e);
  };

  handleLogin = () => {
    const { values, loginAction } = this.props;
    const { username, password } = values;
    loginAction(username, password, history);
  };

  handleSubmit = async e => {
    e.preventDefault();
    const { validateForm, setFieldTouched } = this.props;
    const errors = await validateForm();

    if (isEmpty(errors)) {
      this.handleLogin();
    } else {
      Object.keys(errors).forEach(error => setFieldTouched(error));
    }
  };

  render() {
    const { touched, errors, handleBlur } = this.props;
    const { username, password } = this.props.values;

    return (
      <div className={s.container}>
        <div className={s.login}>
          <form onSubmit={this.handleSubmit}>
            <h1 className={s.title}>{LOGIN_TITLE}</h1>
            <Input
              id="username"
              autoFocus
              label={USERNAME_PLACEHOLDER}
              name="username"
              value={username}
              onChange={this.handleChange}
              onBlur={handleBlur}
              touched={errors.username && touched.username}
              errorMsg={errors.username}
            />

            <Input
              id="password"
              label={PASSWORD_PLACEHOLDER}
              value={password}
              onChange={this.handleChange}
              name="password"
              type="password"
              onBlur={handleBlur}
              touched={errors.password && touched.password}
              errorMsg={errors.password}
            />
            <div className={s.groupBtn}>
              <Button onClick={this.handleSubmit}>{LOGIN_BUTTON}</Button>
              <input className={s.submitBtn} type="submit" />
            </div>
          </form>
        </div>
      </div>
    );
  }
}

const enhance = compose(
  connect(
    null,
    { loginAction: login, logout },
  ),
  withFormik({
    mapPropsToValues: () => ({ username: '', password: '' }),
    validationSchema: getValidationSchema(['username', 'password']),
  }),
  withStyles(s),
);

export default enhance(Login);
