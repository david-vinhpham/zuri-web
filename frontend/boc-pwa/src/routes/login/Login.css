/**
 * BOC VN (http://www.bocvietnam.com/)
 *
 * Copyright Â© 2019-present BOCVN, LLC. All rights reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE.txt file in the root directory of this source tree.
 */
@import '../../components/variables.css';

.container {
  width: 100%;
  height: 100%;
  position: absolute;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: flex-start;
  background: radial-gradient(circle, rgba(255, 255, 255, 1) 0%, rgba(147, 226, 255, 1) 100%);
}

.welcome {
  color: var(--c-brand-3);
  text-align: center;
}

.logo {
  margin: var(--size-64) auto;
  width: var(--home-logo-width);
}

.title {
  margin: var(--size-12) 0 var(--size-36);
}

.brand {
  width: var(--size-86);
  margin: var(--size-14);
}

.greeting {
  color: var(--c-brand-1);
  font-size: var(--font-size-16);
}

.facebookInitAK {
  display: flex;
  position: absolute;
  bottom: var(--size-64);
}

.button {
  color: var(--c-text-bright);
  background: var(--c-brand-1);
}

.copyRight {
  position: absolute;
  bottom: var(--size-12);
  color: var(--c-brand-1);
  font-size: var(--font-size-10);
  text-align: center;
}
