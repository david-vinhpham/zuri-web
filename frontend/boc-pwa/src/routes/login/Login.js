/**
 * BOC VN (http://www.bocvietnam.com/)
 *
 * Copyright Â© 2018-present BOCVN, LLC. All rights reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE.txt file in the root directory of this source tree.
 */

import React from 'react';
import PropTypes from 'prop-types';
import withStyles from 'isomorphic-style-loader/lib/withStyles';
import BocLogo from './images/boc_logo.png';
import s from './Login.css';

class Login extends React.Component {
  static propTypes = {
    title: PropTypes.string.isRequired,
  };

  render() {
    return (
      <div className={s.root}>
        <div className={s.container}>
          <h1>{this.props.title}</h1>
          <img src={BocLogo} alt="boc-logo" className={s.logo} />
        </div>
      </div>
    );
  }
}

export default withStyles(s)(Login);
