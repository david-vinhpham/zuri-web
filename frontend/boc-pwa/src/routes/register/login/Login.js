import React, { PureComponent } from 'react';
import { Input, Button } from 'homecredit-ui';
import { withFormik } from 'formik';
import { compose } from 'redux';
import { connect } from 'react-redux';
import { func, shape, bool, string } from 'prop-types';
import withStyles from 'isomorphic-style-loader/lib/withStyles';
import noop from 'lodash/noop';
import isEmpty from 'lodash/isEmpty';

import { getValidationSchema } from 'utils/validations';
import { login, clearUserInfo } from 'actions/saRegister';
import s from './Login.css';

import {
  LOGIN_TITLE,
  USERNAME_PLACEHOLDER,
  PASSWORD_PLACEHOLDER,
  LOGIN_BUTTON,
} from './constant';

class Login extends PureComponent {
  static propTypes = {
    values: shape({
      saUsername: string,
      saPassword: string,
    }),
    loginAction: func,
    touched: shape({
      saUsername: bool,
      saPassword: bool,
    }),
    errors: shape({
      saUsername: string,
      saPassword: string,
    }),
    clearUserInfoAction: func,
    handleChange: func,
    handleBlur: func,
    validateForm: func.isRequired,
    setFieldTouched: func.isRequired,
  };

  static defaultProps = {
    values: {},
    loginAction: noop,
    clearUserInfoAction: noop,
    touched: {},
    errors: {},
    handleChange: noop,
    handleBlur: noop,
  };

  /**
   * clear sa token when user enter route
   */
  componentDidMount = () => {
    this.props.clearUserInfoAction();
  };

  handleChange = e => {
    const { handleChange } = this.props;
    const { name, value } = e.target;

    if (name === 'saUsername' && /[^a-zA-Z0-9]/.test(value)) return;

    handleChange(e);
  };

  handleLogin = () => {
    const { values, loginAction } = this.props;
    const { saUsername, saPassword } = values;
    loginAction(saUsername, saPassword);
  };

  handleSubmit = async e => {
    e.preventDefault();
    const { validateForm, setFieldTouched } = this.props;
    const errors = await validateForm();

    if (isEmpty(errors)) {
      this.handleLogin();
    } else {
      Object.keys(errors).forEach(error => setFieldTouched(error));
    }
  };

  render() {
    const { touched, errors, handleBlur } = this.props;
    const { saUsername, saPassword } = this.props.values;

    return (
      <div className={s.container}>
        <div className={s.login}>
          <form onSubmit={this.handleSubmit}>
            <h2 className={s.title}>{LOGIN_TITLE}</h2>
            <Input
              id="saUsername"
              label={USERNAME_PLACEHOLDER}
              name="saUsername"
              value={saUsername}
              onChange={this.handleChange}
              onBlur={handleBlur}
              touched={errors.saUsername && touched.saUsername}
              errorMsg={errors.saUsername}
              msgPosition="below"
              maxLength={20}
            />

            <Input
              id="saPassword"
              label={PASSWORD_PLACEHOLDER}
              value={saPassword}
              onChange={this.handleChange}
              name="saPassword"
              type="password"
              onBlur={handleBlur}
              touched={errors.saPassword && touched.saPassword}
              errorMsg={errors.saPassword}
              msgPosition="below"
            />
          </form>
          <div className={s.groupBtn}>
            <Button onClick={this.handleSubmit}>{LOGIN_BUTTON}</Button>
          </div>
        </div>
      </div>
    );
  }
}

const enhance = compose(
  connect(
    null,
    dispatch => ({
      loginAction: (username, password) => dispatch(login(username, password)),
      clearUserInfoAction: () => dispatch(clearUserInfo()),
    }),
  ),
  withFormik({
    mapPropsToValues: () => ({ saUsername: '', saPassword: '' }),
    validationSchema: getValidationSchema(['saUsername', 'saPassword']),
  }),
  withStyles(s),
);

export default enhance(Login);
