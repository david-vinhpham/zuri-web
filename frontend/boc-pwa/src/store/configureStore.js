import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import { composeWithDevTools } from 'redux-devtools-extension/developmentOnly';
import { persistStore, persistReducer } from 'redux-persist';

import autoMergeLevel2 from 'redux-persist/lib/stateReconciler/autoMergeLevel2';

import storage from 'redux-persist/lib/storage';
import { name, version } from '../../package.json';
import rootReducer from '../reducers';
import createLogger from './logger';

export default function configureStore(initialState, helpersConfig) {
  const middleware = [thunk.withExtraArgument(helpersConfig)];

  const persistConfig = {
    key: 'BOCVN',
    storage,
    stateReconciler: autoMergeLevel2,
    blacklist: ['clz'],
  };

  const persistedReducer = persistReducer(persistConfig, rootReducer);

  let enhancer;

  if (__DEV__) {
    middleware.push(createLogger());

    // https://github.com/zalmoxisus/redux-devtools-extension#14-using-in-production
    const composeEnhancers = composeWithDevTools({
      // Options: https://github.com/zalmoxisus/redux-devtools-extension/blob/master/docs/API/Arguments.md#options
      name: `${name}@${version}`,
    });

    // https://redux.js.org/docs/api/applyMiddleware.html
    enhancer = composeEnhancers(applyMiddleware(...middleware));
  } else {
    enhancer = applyMiddleware(...middleware);
  }

  // https://redux.js.org/docs/api/createStore.html
  const store = createStore(persistedReducer, enhancer);

  // Hot reload reducers (requires Webpack or Browserify HMR to be enabled)
  if (__DEV__ && module.hot) {
    module.hot.accept('../reducers', () =>
      // Don't forget to remove `()` if you change reducers back to normal rootReducer.
      // eslint-disable-next-line global-require
      store.replaceReducer(require('../reducers').default()),
    );
  }

  const persistor = persistStore(store);

  return { store, persistor };
}
