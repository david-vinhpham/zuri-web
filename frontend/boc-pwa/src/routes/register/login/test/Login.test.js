import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';

import Login from '../Login';

const mockStore = configureStore();
const initialState = {
  saRegister: {
    auth: {
      username: null,
    },
  },
};

it('Login snapshot', () => {
  const store = mockStore(initialState);
  const wrapper = renderer.create(
    <Provider store={store}>
      <Login />
    </Provider>,
  );
  const snapshot = wrapper.toJSON();

  expect(snapshot).toMatchSnapshot();
});
