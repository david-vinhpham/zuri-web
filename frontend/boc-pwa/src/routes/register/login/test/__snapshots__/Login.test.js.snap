// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VerifyCode snapshot 1`] = `
<div
  className="container"
>
  <div
    className="login"
  >
    <form
      onSubmit={[Function]}
    >
      <h2
        className="title"
      >
        Thông tin đăng nhập
      </h2>
      <div
        className="Input_wrapper__WALTg"
      >
        <i
          className="material-icons Input_icon__1jjsS"
        >

        </i>
        <input
          accept=""
          autoComplete=""
          className="Input_input__1IXXH"
          disabled={false}
          id="saUsername"
          maxLength={20}
          name="saUsername"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onMouseDown={[Function]}
          placeholder=""
          readOnly={false}
          type="text"
          value=""
        />
        <label
          className="Input_label__2ZxWx"
          htmlFor="saUsername"
        >
          Tên đăng nhập
        </label>
        <div
          className="Input_underline__Xwqb3"
        />
      </div>
      <div
        className="Input_wrapper__WALTg"
      >
        <i
          className="material-icons Input_icon__1jjsS"
        >

        </i>
        <input
          accept=""
          autoComplete=""
          className="Input_input__1IXXH"
          disabled={false}
          id="saPassword"
          maxLength={null}
          name="saPassword"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onMouseDown={[Function]}
          placeholder=""
          readOnly={false}
          type="password"
          value=""
        />
        <label
          className="Input_label__2ZxWx"
          htmlFor="saPassword"
        >
          Mật khẩu
        </label>
        <div
          className="Input_underline__Xwqb3"
        />
      </div>
    </form>
    <div
      className="groupBtn"
    >
      <button
        className="Button_buttonReset__3cQTl Button_button__2pQRN"
        disabled={false}
        onClick={[Function]}
        title=""
        type="button"
      >
        Đăng nhập
      </button>
    </div>
  </div>
</div>
`;
